extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -268,23 +268,38 @@ class MongoQueueProcessor {
      * Update ingested entry metadata location field. Each location change
      * includes: key, dataStoreName, dataStoreType, dataStoreVersionId
      * @param {ObjectQueueEntry} entry - object queue entry object
-     * @param {string} location - zenko storage location name
+     * @param {string} zenkoLocation - zenko storage location name
      * @return {undefined}
      */
-    _updateLocations(entry, location) {
+    _updateLocations(entry, zenkoLocation) {
         const locations = entry.getLocation();
-        const editLocations = locations.map(l => {
-            const newValues = {
+        if (!locations || locations.length === 0) {
+            const dataStoreVersionId = entry.getVersionId() ?
+                entry.getEncodedVersionId() : '';
+            const editLocation = [{
                 key: entry.getObjectKey(),
-                dataStoreName: location,
+                size: 0,
+                start: 0,
+                dataStoreName: zenkoLocation,
                 dataStoreType: 'aws_s3',
-            };
-            if (entry.getVersionId()) {
-                newValues.dataStoreVersionId = entry.getEncodedVersionId();
-            }
-            return Object.assign({}, l, newValues);
-        });
-        entry.setLocation(editLocations);
+                dataStoreETag: `1:${emptyFileMd5}`,
+                dataStoreVersionId,
```

Comment:
We don't seem to treat the same way the `dataStoreVersionId` here and below when there are locations which looks odd, setting it to the empty string here when no version is present, not setting it at when there are locations.

Reply:
ah yea good catch, below I should set to empty string